docs/rules/object-key-newlines.md

```diff
@@ -0,0 +1,68 @@
+# Enforce placing object keys on separate lines (object-key-newlines)
+
+While formatting preferences are very personal, a number of style guides require that
```

Comment:
@nzakas @pedrottimark I think the arguments for this style are:
- It improves code readability greatly
- It also improves readability of diffs. An example comparison of both cases when modifying a single property: 

```diff
 var foo = {
     foo: 'foo',
-    bar: 'bar',
+    bar: 'bazz',
     qux: 'qux'
 };
```

```diff
-var foo = { foo: 'foo', bar: 'bar', qux: 'qux' };
+var foo = { foo: 'foo', bar: 'bazz', qux: 'qux' };
```

Anything else you think we should include?